src/dsl/dynamic.jl

```diff
@@ -8,19 +8,21 @@ function make_dynamic_gen_function(name, args, body, return_type, annotations)
     julia_fn_defn = Expr(:function,
         Expr(:call, esc(julia_fn_name), gf_args...),
         esc(body))
-
-    # create generator and assign it a name
     arg_types = map((arg) -> esc(arg.typ), args)
     has_argument_grads = map(
         (arg) -> (DSL_ARG_GRAD_ANNOTATION in arg.annotations), args)
     accepts_output_grad = DSL_RET_GRAD_ANNOTATION in annotations
-    Expr(:block,
-        julia_fn_defn,
-        Expr(:(=), 
-            esc(name),
-            Expr(:call, :DynamicDSLFunction,
-                quote Type[$(arg_types...)] end,
-                julia_fn_defn,
-                has_argument_grads, esc(return_type),
-                accepts_output_grad)))
+
+    quote
```

Comment:
Checking my understanding: Does this quoted bit get recursively parsed, where the recursive parsing is needed because the `Expr(:call, ...)` on the left does not recursively expand macros? (And in our case we need to recursively expand the macro `Core.@__doc__`)

Reply:
The quote isn't needed. It would have been possible to construct the AST directly as before. In this commit, I shifted to using `quote` because (i) it gives more readable cleaner code, and (ii) when I tried to put `Core.@__doc__` into the AST (it involves creating a `Expr(:macrocall, <macroname>, ...)` AST node (see https://docs.julialang.org/en/v1/devdocs/ast/index.html#Macros-1) it didn't work immediately, so it just seemed simpler to use `quote .. end`, in which case it's Julia's responsibility to construct the properly formed AST.

Reply:
> it involves creating a `Expr(:macrocall, <macroname>, ...`

Ah, got it.